File: z-clients/javascript/src/models/adminAccessPolicyOut.ts
// this file is @generated
import {
    type AccessRule,
    AccessRuleSerializer,
} from './accessRule';

export interface AdminAccessPolicyOut {
    id: string;
    description: string;
    rules: AccessRule[];
    created: Date;
    updated: Date;
}

export const AdminAccessPolicyOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AdminAccessPolicyOut {
        return {
            id: object['id'],
            description: object['description'],
            rules: object['rules'].map((item: AccessRule) => AccessRuleSerializer._fromJsonObject(item)),
            created: new Date(object['created']),
            updated: new Date(object['updated']),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AdminAccessPolicyOut): any {
        return {
            'id': self.id,
            'description': self.description,
            'rules': self.rules.map((item) => AccessRuleSerializer._toJsonObject(item)),
            'created': self.created,
            'updated': self.updated,
        };
    }
}

File: z-clients/rust/src/models/admin_access_policy_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::access_rule::AccessRule;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AdminAccessPolicyOut {
    pub id: String,

    pub description: String,

    pub rules: Vec<AccessRule>,

    #[serde(with = "crate::unix_timestamp_ms_serde")]
    pub created: jiff::Timestamp,

    #[serde(with = "crate::unix_timestamp_ms_serde")]
    pub updated: jiff::Timestamp,
}

impl AdminAccessPolicyOut {
    pub fn new(
        id: String,
        description: String,
        rules: Vec<AccessRule>,
        created: jiff::Timestamp,
        updated: jiff::Timestamp,
    ) -> Self {
        Self {
            id,
            description,
            rules,
            created,
            updated,
        }
    }
}
